### DISCUSSION TITLE
v0.1

### DISCUSSION THREAD

**Discussion by chekos** [chekos/pypums#277] [General]
## What’s Changed

* Update docs (#271) @chekos
* Major restructuring of the entire project.

## :rocket: Features

* Moving to a much simpler project structure. (#269) @chekos

## :arrow_up: Dependencies updates

* :arrow_up: Bump actions/setup-python from 2 to 3 (#274) @dependabot
* :arrow_up: Bump actions/checkout from 2 to 3 (#275) @dependabot

## :busts_in_silhouette: List of contributors

@chekos

## Breaking changes
* `ACS().download_data()` -> `ACS().download()`
* The `person_or_household` argument has been renamed to the more appropriate `sample_unit`
* `pypums` will now default to downloading data to a specified folder using `click`'s `get_app_dir()`. More info [here](https://click.palletsprojects.com/en/8.0.x/api/?highlight=get_app_dir#click.get_app_dir)
* `pypums` now checks if a dataset has been previously downloaded and will default to _not_ downloading it unless the `overwrite` argument is set to `True`
* the `pypums` CLI now has two commands:
  * `acs-url` which builds the URL to download data from the Census server ([docs](https://pypums.readthedocs.io/en/latest/reference/cli/#acs-url))
  * `download-acs` which takes `--year`, `--state`, `--sample-unit` and `--survey` options (and [more](https://pypums.readthedocs.io/en/latest/reference/cli/#download-acs)) to download (and extract, by default) the data.
* We moved to `httpx` instead of `requests`
* We moved to `rich` instead of `tqdm` for progress bars
* We've added tests 😅 

<hr /><em>This discussion was created from the release <a href='https://github.com/chekos/pypums/releases/tag/v0.1'>v0.1</a>.</em>

**Comment by akhiljr10** [comment]
Can you also add option to support the use of 2020 ACS data that recently came out?

**Comment by chekos** [comment]
Absolutely! The 2020 data is only experimental because of the impacts of COVID (tracking it in #278)

So just need to work on some way to flag it as experimental as opposed to the rest of the surveys.

**Reply by akhiljr10** [reply]
Absolutely agree with that, but as ACS has the data available it would be beneficial for users working on it to directly pull it from there rather than downloading the file into the local system.
